core-web/libs/sdk/client/src/lib/query-builder/sdk-query-builder.spec.ts

```diff
@@ -0,0 +1,155 @@
+import { QueryBuilder } from './sdk-query-builder';
+
+describe('QueryBuilder', () => {
+    let queryBuilder: QueryBuilder;
+
+    beforeEach(() => {
+        queryBuilder = new QueryBuilder();
+    });
+
+    it('should return a query with a simple term', () => {
+        const queryForBlogs = queryBuilder.field('contentType').term('Blog').build();
+
+        expect(queryForBlogs).toBe('+contentType:Blog');
+    });
+
+    it('should return a query with multiple fields with a simple term ', () => {
+        const queryForBlogsInSuperCoolSite = queryBuilder
+            .field('contentType')
+            .term('Blog')
+            .field('conhost')
+            .term('my-super-cool-site')
+            .build();
+
+        expect(queryForBlogsInSuperCoolSite).toBe('+contentType:Blog +conhost:my-super-cool-site');
+    });
+
+    it("should return a query with an 'OR' operand", () => {
+        const queryForBlogsOrArticles = queryBuilder
+            .field('contentType')
+            .term('Blog')
+            .or()
+            .term('Article')
+            .build();
+
+        expect(queryForBlogsOrArticles).toBe('+contentType:Blog OR Article');
+    });
+
+    it("should return a query with an 'AND' operand", () => {
+        const queryForBlogsAndArticles = queryBuilder
+            .field('contentType')
+            .term('Blog')
+            .and()
+            .term('Article')
+            .build();
+
+        expect(queryForBlogsAndArticles).toBe('+contentType:Blog AND Article');
+    });
+
+    it("should return a query with a 'NOT' operand", () => {
+        const queryForSkiingTripsNotInSwissAlps = queryBuilder
```

Comment:
I like the idea of the Query Builder, but after usign in the past others like CakePHP ORM/ Laravel (in PHP) or Prisma in TS, I feel if we are going to do something that need to be easy to use  or explicit  what are we doing only to see the builder, Example:

If I want a contentType `Blog` or `Article` do something more redeable.
```
const query = queryBuilder
    .find(). // find in few orm is a only one item, and findAll is for an array
    .where({ 'contentType': 'Blog' })
    .orWhere({ 'contentType': 'Article' })
```

Or

```
const query = queryBuilder
    .find()
    .where({ 'summary': 'Skiing trip' })
    .notWhere({ 'summary': 'Swiss Alps' })
```

From
```
const snowboardInCanada = anotherQueryBuilder
            .field('summary')
            .term('Snowboard')
            .field('country')
            .term('Canada')
            .build();
```
To
```
const snowboardInCanada  = anotherQueryBuilder
            .find()
            .where([
                        {'summary': 'Snowboard'}, 
                        {'country': 'Canada'}
             ]) // send one object or an array of it.
```

Should be more intuitive use more the words, `select`, `find`,  `where` etc.

Reply:
I think those ideas are good but it should be a diff abstraction depending on the level, for me the queryBuilder is a low-level resource, higher level resource will encapsulate the complexity of the query builder, but if someone needs more control can always use the query builder or even a raw query it will depend on the user case

Reply:
This is cool, but I don't know if it's what we want, the migration to that should not be difficult

Reply:
Also Lucene Queries could be verycomplex and the way that I made this handles that complexity in a low level way, I think if we stick to that kind of queries it could be a limit to the developer when trying to build complex queries

Reply:
Following the same pattern:
What about merge `field` and `term`:
from this:
```
const queryForFootballBlogsWithoutMessi = queryBuilder
            .field('contentType')
            .term('Blog')
            .field('title')
            .term('Football')
            .excludeField('summary')
            .term('Lionel Messi')
            .build();
```

to this
```
const queryForFootballBlogsWithoutMessi = queryBuilder
            .field('contentType', 'Blog')
            .field('title', 'Football')
            .exclude('summary', 'Lionel Messi')
            .build();
```

Reply:
Following @jdotcms ideas, maybe we can keep the `QueryBuilder` closer to Lucene Query, and we can have another level closers to what @oidacra is saying? I don't link to much the `where` or `orWhere`, I don't want people to think this is `SQL` or an `ORM` to interact with dotCMS database.